src/libpgmoneta/restore.c

```diff
@@ -818,6 +818,108 @@ pgmoneta_combine_backups(int server, char* label, char* base, char* input_dir, c
    return 1;
 }
 
+int
+pgmoneta_backup_size_incremental(int server, char* label, unsigned long* size, uint64_t* biggest_file_size)
+{
+   struct art* nodes = NULL;
+   struct backup* backup = NULL;
+   struct json* manifest_read = NULL;
+   struct json* files = NULL;
+   struct main_configuration* config = NULL;
+   struct json_iterator *iter = NULL;
+   char* manifest_path = NULL;
+   unsigned long sz = 0;
+   uint64_t biggest_file_sz = 0;
+
+   config = (struct main_configuration*)shmem;
+   pgmoneta_art_create(&nodes);
+   if (pgmoneta_workflow_nodes(server, label, nodes, &backup))
+   {
+      goto error;
+   }
+
+   // read and traverse the manifest of the incremental backup
+   manifest_path = pgmoneta_get_server_backup_identifier_data(server, label);
+   manifest_path = pgmoneta_append(manifest_path, "backup_manifest");
+   if (pgmoneta_json_read_file(manifest_path, &manifest_read))
+   {
+      pgmoneta_log_error("Unable to read manifest %s", manifest_path);
+      goto error;
+   }
+
+   files = (struct json*)pgmoneta_json_get(manifest_read, MANIFEST_FILES);
+
+   if (files == NULL)
+   {
+      goto error;
+   }
+
+   pgmoneta_json_iterator_create(files, &iter);
+   while (pgmoneta_json_iterator_next(iter))
+   {
+      struct json* file = NULL;
+      char* file_path = NULL;
+      uint64_t file_size = 0;
+
+      file = (struct json*)pgmoneta_value_data(iter->value);
+      file_path = (char*)pgmoneta_json_get(file, "Path");
+      if (pgmoneta_contains(file_path, INCREMENTAL_PREFIX)) // for incremental files get the `truncated_block_length`
```

Comment:
Use `pgmoneta_is_incremental_path()`, in case (though unlikely) some path has incremental prefix somewhere in the middle